arm/Microsoft.Storage/storageAccounts/.bicep/nested_container.bicep

```diff
@@ -20,11 +20,11 @@ resource container_policy 'Microsoft.Storage/storageAccounts/blobServices/contai
   ]
 }
 
-module container_rbac './nested_container_rbac.bicep' = [for (roleassignment, index) in blobContainer.roleAssignments: {
+module container_rbac './nested_container_rbac.bicep' = [for (roleAssignment, index) in blobContainer.roleAssignments: {
   name: '${deployment().name}-Rbac-${(empty(blobContainer.roleAssignments) ? 'dummy' : index)}'
   params: {
     blobContainerName: blobContainer.name
-    roleAssignment: roleassignment
+    roleAssignmentObj: roleAssignment
     builtInRoleNames: builtInRoleNames
     storageAccountName: storageAccountName
   }
```

Comment:
Is this dependency needed?

Reply:
If we want to fix the depensOn, any of the fields of the resource would need to reference another field from i.e `container`. So if line 26 could be `blobContainerName: container.name` and do the same changes in nested_container_rbac.bicep (removing the storageAccount as param and use the blobContainerName in the name field, then yes you could remove dependency. Its only going to map dependency if we reference a field from another resource.